subsys/infinispan/src/main/resources/infinispan.xml

```diff
@@ -1,75 +1,70 @@
       </persistence>
     </local-cache>
 
-    <local-cache name="content-index" configuration="local-template">
-      <eviction strategy="LRU" size="200000" type="COUNT"/>
-    </local-cache>
+    <local-cache name="content-metadata" configuration="local-template"/>
+
+    <local-cache name="content-index" configuration="local-template"/>
 
     <local-cache name="maven-version-metadata-cache" deadlock-detection-spin="10000" configuration="local-template">
-      <eviction size="10000000" type="COUNT" strategy="LRU"/>
+      <memory>
+        <object size="10000000"/>
+      </memory>
     </local-cache>
 
     <local-cache name="indy-nfs-owner-cache" deadlock-detection-spin="10000" configuration="local-template">
-      <eviction size="200000" type="COUNT" strategy="LRU"/>
-      <transaction transaction-manager-lookup="org.infinispan.transaction.lookup.DummyTransactionManagerLookup"
-                   locking="PESSIMISTIC"/>
+      <transaction protocol="DEFAULT" transaction-manager-lookup="org.infinispan.transaction.lookup.DummyTransactionManagerLookup" locking="PESSIMISTIC"/>
     </local-cache>
 
-    <!--
-        This cache works for delete the fast local cache of the NFS supported repo cache on local. With the expiration,
-        it will make all cache entries expired after 1 day, and trigger the purge of the expired cache every 30 mins
-    -->
     <local-cache name="indy-fastlocal-file-delete-cache" configuration="local-template">
-      <eviction size="200000" strategy="LRU"/>
+      <!--
+        Delete the fast local cache entries. It expires entries in 1 day and run expiration every 30 minutes.
+      -->
       <expiration lifespan="86400000" max-idle="86400000" interval="1800000"/>
     </local-cache>
 
     <local-cache name="schedule-expire-cache" configuration="local-template">
-      <!--
-          this expiration interval is used to trigger the purge threads of ISPN to let it purge the cache, which will
-          trigger the cache expire event to happen. Did not find a way to let the expire event automatically happen
-          without this. See http://infinispan.org/docs/stable/faqs/faqs.html#eviction_and_expiration_questions for more
-       -->
-      <expiration interval="300" />
+      <expiration interval="300000" /><!-- run expiration every 5 minutes -->
     </local-cache>
 
     <local-cache name="nfc" configuration="local-template">
       <!--
-          Use both eviction and expiration to guide against OOM, and run expiration thread twice an hour. Refer to
-          http://infinispan.org/docs/8.2.x/user_guide/user_guide.html#configuration
+        Delete NFC cache entries. It expires entries in 1 hour and run expiration every 15 minutes.
       -->
-      <eviction size="10000000" type="COUNT" strategy="LRU"/>
-      <expiration interval="1800" />
+      <expiration lifespan="3600000" max-idle="3600000" interval="900000" />
```

Comment:
We need to override the NFC cache config in the custom infinispan.xml to use off-heap memory. (as in the test config file above)